source/common/grpc/common.cc

```diff
@@ -148,18 +148,39 @@ Buffer::InstancePtr Common::serializeBody(const Protobuf::Message& message) {
 
 Http::MessagePtr Common::prepareHeaders(const std::string& upstream_cluster,
                                         const std::string& service_full_name,
-                                        const std::string& method_name) {
+                                        const std::string& method_name,
+                                        const absl::optional<std::chrono::milliseconds>& timeout) {
   Http::MessagePtr message(new Http::RequestMessageImpl());
   message->headers().insertMethod().value().setReference(Http::Headers::get().MethodValues.Post);
   message->headers().insertPath().value().append("/", 1);
   message->headers().insertPath().value().append(service_full_name.c_str(),
                                                  service_full_name.size());
   message->headers().insertPath().value().append("/", 1);
   message->headers().insertPath().value().append(method_name.c_str(), method_name.size());
+  message->headers().insertTE().value().setReference(Http::Headers::get().TEValues.Trailers);
   message->headers().insertHost().value(upstream_cluster);
+  if (timeout) {
+    uint64_t time = timeout.value().count();
```

Comment:
extract this part to a function like `std::string toGrpcTimeout(std::chrono::milliseconds& timeout)`.

Reply:
Added `static void toGrpcTimeout(const std::chrono::milliseconds& timeout, Http::HeaderString& value)` as it is not needed from anywhere else and to avoid a temporary std::string.